tuxedo-core/aggregator/src/lib.rs

```diff
@@ -115,12 +130,28 @@ pub fn tuxedo_constraint_checker(attrs: TokenStream, body: TokenStream) -> Token
     let variants = variant_type_pairs.clone().map(|(v, _t)| v);
     let inner_types = variant_type_pairs.map(|(_v, t)| t);
 
-    let vis = ast.vis;
+    // Set up the names of the new associated types.
     let mut error_type_name = outer_type.to_string();
     error_type_name.push_str("Error");
     let error_type = Ident::new(&error_type_name, outer_type.span());
-    let inner_types = inner_types.clone();
+
+    let mut inherent_hooks_name = outer_type.to_string();
+    inherent_hooks_name.push_str("InherentHooks");
+    let inherent_hooks = Ident::new(&inherent_hooks_name, outer_type.span());
+
+    let vis = ast.vis;
+
+    // TODO there must be a better way to do this, right?
```

Comment:
I'm fine keeping this ugly todo for now. But borrowing is a good idea. I honestly didn't think of that. Let me try it.

Reply:
`cannot move out of `*variants` which is behind a shared reference`

Since none of us are macro experts and it is working right now, I'm going to leave it. Hopefully someday we'll get a real macro expert to take a look.